tests/test_data.py

```diff
@@ -1506,7 +1506,7 @@ async def insert_resource(env_id, version, agent_name, path, status):
     assert len(result) == 3
     assert sorted([r["agent"] for r in result]) == ["agent1", "agent1", "agent2"]
     for r in result:
-        assert len(r["attributes"]) == 1
+        assert len(r["attributes"]) == 2  # path and version attribute
```

Comment:
Could you elaborate on what exactly changed here? I got the impression we inserted the version field for backwards compatibility, but this assertion seems to imply otherwise.

Reply:
When the exporter sends resources to the server, the version field is included in two different locations. Once in the version field of the dictionary and once in the `attributes["version"]` field of the dictionary. It was decided that we want to keep the API backwards compatible, but that we don't want to include the redundant version data into the database. The problem is that when we remove the version field from the attributes dictionary in the database, we have no way of knowing whether the version field was there when the resources were sent to the server. This is not an issue in practice because the exporter always does, but this test case doesn't. Maybe I should have fixed this test case the other way around, namely adding the version field to the attributes dictionary.

Reply:
I understand. We don't use the exporter here so we never send the version field. We used to get back exactly what we sent in, but now the version field is added in the layer above the database because we don't persist it anymore, so we always get it out, even if we didn't include it in the insert.